### PR TITLE
spotify-api edit RecommendationsFromSeedsResponse and AlbumObjectSimplified

### PR DESCRIPTION
**Pull request by sds-smith**
Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x] Test the change in your own code. (Compile and run.)
- [ x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

** Edits interface RecommendationsObject to replace TrackObjectSimplified[] with TrackObjectFull[]; edits interface AlbumObjectSimplified.album_type to include values in all caps as returned by the recommendations endpoint.

Screenshots provided:
  * VSCode editor showing ts error when using SpotifyApi.RecommendationsFromSeedsResponse (no album property on TrackObjectSimplified)
  * VSCode editor showing no ts error when using updated type with TrackObjectFull[]
  * console.log of Spotify response in the form of TrackObjectFull[]
  * 
console log_RecommendationsResponse
RecommendationsFromSeedsResponse
with_TrackObjectFull _type

### CODE REVIEW

File: types/spotify-api/index.d.ts [hunk 1]
```diff
@@ -874,7 +874,8 @@ declare namespace SpotifyApi {
         /**
          * The type of the album: one of “album”, “single”, or “compilation”.
          */
-        album_type: 'album' | 'single' | 'compilation';
+        album_type: 'album' | 'single' | 'compilation' |
+                    'ALBUM' | 'SINGLE' | 'COMPILATION';
```
Comment (piksel) [inline types/spotify-api/index.d.ts:878]: I think you found a "bug" in the spotify API. Requesting the same album returns the "album_type" in all lower case:
```
GET https://api.spotify.com/v1/albums/6wGcuJq8QjSeoYMJWgW7Wo
Authorization: Bearer {{access_token}}
Content-Type: application/json

###

{
  "album_type": "single",
  "artists": [
    {
      "external_urls": {
        "spotify": "https://open.spotify.com/artist/7jy3rLJdDQY21OgRLCZ9sD"
      },
      "href": "https://api.spotify.com/v1/artists/7jy3rLJdDQY21OgRLCZ9sD",
      "id": "7jy3rLJdDQY21OgRLCZ9sD",
      "name": "Foo Fighters",
[...]
```
But requesting the `/recommendations` yields the `album_type` in caps. I suggest adding a separate type for recommendations, as that seems to be the only endpoint that returns the type in this manner.

File: types/spotify-api/index.d.ts [hunk 2]
```diff
@@ -1293,7 +1294,7 @@ declare namespace SpotifyApi {
      */
     interface RecommendationsObject {
         seeds: RecommendationsSeedObject[];
-        tracks: TrackObjectSimplified[];
+        tracks: TrackObjectFull[];
     }
```
Comment (piksel) [inline types/spotify-api/index.d.ts:1298]:
```suggestion
        tracks: RecommendationTrackObject[];
    }
    
    /**
     * Recommendation Track Object
     * Uses the same object structure as Full Track Object, but with `album.album_type` in caps.
     */
    interface RecommendationTrackObject extends Omit<TrackObjectFull, "album"> {
	    album: RecommendationAlbumObject;
    }

    /**
     * Recommendation Album Object
     * Uses the same object structure as Simple Album Object, but with `album_type` in caps.
     */
    interface RecommendationAlbumObject extends Omit<AlbumObjectSimplified, "album_type"> {
        /**
         * The type of the album: one of “ALBUM”, “SINGLE”, or “COMPILATION”.
	     * Note that this differs from the types returned by all other spotify APIs by being in all caps.
         */
        album_type: 'ALBUM' | 'SINGLE' | 'COMPILATION';
    }
```

**Review by piksel — CHANGES_REQUESTED** [review]
The extra fields in the recommendations are not documented by Spotify and does not seem to align with the rest of the API. Added a suggestion for typing it so that it does not interfere with the rest of the API endpoints.

**Comment by sds-smith** [comment]
Thank you. Committed suggested revisions

File: types/spotify-api/index.d.ts (hunk 2 above)
Comment (sds-smith) [reply types/spotify-api/index.d.ts]: Receiving the following error in test logs after selecting 'Commit suggestion': 
ERROR: 1395:1  jsdoc-format  asterisks in jsdoc must be aligned

Not sure how to resolve, as this and surrounding lines appear to comply with jsdoc-format rules.

File: types/spotify-api/index.d.ts [hunk 3]
```diff
@@ -1373,7 +1374,27 @@ declare namespace SpotifyApi {
...
-        tracks: TrackObjectSimplified[];
+        tracks: RecommendationTrackObject[];
+    }
+    
+    /**
+     * Recommendation Track Object
+     * Uses the same object structure as Full Track Object, but with `album.album_type` in caps.
+     */
+    interface RecommendationTrackObject extends Omit<TrackObjectFull, "album"> {
+	    album: RecommendationAlbumObject;
```
Comment (piksel) [inline types/spotify-api/index.d.ts:1385]:
```suggestion
        album: RecommendationAlbumObject;
```

File: types/spotify-api/index.d.ts [hunk 4]
```diff
@@ -1373,7 +1374,27 @@ declare namespace SpotifyApi {
...
+    }
+
+    /**
+     * Recommendation Album Object
+     * Uses the same object structure as Simple Album Object, but with `album_type` in caps.
+     */
+    interface RecommendationAlbumObject extends Omit<AlbumObjectSimplified, "album_type"> {
+        /**
+         * The type of the album: one of “ALBUM”, “SINGLE”, or “COMPILATION”.
+	     * Note that this differs from the types returned by all other spotify APIs by being in all caps.
```
Comment (piksel) [inline types/spotify-api/index.d.ts]:
```suggestion
         * The type of the album: one of “ALBUM”, “SINGLE”, or “COMPILATION”.
         * Note that this differs from the types returned by all other spotify APIs by being in all caps.
```

**Review by piksel — CHANGES_REQUESTED** [review]
There are tab characters instead of spaces in the change. This should fix it.

File: types/spotify-api/index.d.ts (hunk 3 above)
Comment (DangerBotOSS) [inline types/spotify-api/index.d.ts:1385]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/spotify-api/index.d.ts [hunk 5]
```diff
@@ -954,7 +954,8 @@ declare namespace SpotifyApi {
         /**
          * The type of the album: one of “album”, “single”, or “compilation”.
          */
-        album_type: 'album' | 'single' | 'compilation';
+        album_type: 'album' | 'single' | 'compilation' |
+                    'ALBUM' | 'SINGLE' | 'COMPILATION';
```
Comment (piksel) [inline types/spotify-api/index.d.ts:958]:
```suggestion
        album_type: 'album' | 'single' | 'compilation';
```
This should still be reverted, as only the `RecommendationsFromSeeds` returns these types.

**Review by piksel — CHANGES_REQUESTED** [review]

**Review by piksel — APPROVED** [review]

**Comment by piksel** [comment]
Ready to merge

**Pull request merged**